### DISCUSSION TITLE
Type definition for objects that Quasar Components generate. Typescript

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [quasarframework/quasar#11858] [General - Components / Directives / etc] [answered]
Hi, I'd like to know if Quasar(v2) offers some *type definition* for the Javascript object that a Quasar component generates.
Inspecting an element object generated by QHeader I see:
![2022-01-01_22h52_56](https://user-images.githubusercontent.com/15804058/147863966-a72ccd70-9ea7-4e63-9526-329bc272f738.png)

I get TS errors because I don't know what kind of type is this object:
![2022-01-01_22h54_22](https://user-images.githubusercontent.com/15804058/147863984-ba83c666-7f17-48c7-964e-79d8e4d2ee91.png)

Anyone knows if Quasar offers such a type definition? If not, would you plan to add this in the future? Thanks

**Comment by yusufkandemir** [comment, marked as answer]
I would recommend using Composition API, especially when using TypeScript. In this instance, it makes it possible to type template refs. (_I've used script setup to simplify it even further, but it's optional_)

```html
<script lang="ts" setup>
import { ref, Ref } from 'vue';
import { QHeader } from 'quasar';

const headerElementRef = ref() as Ref<QHeader>;

onMounted(() => {
  console.log(headerElementRef.value.heightHint) // Can be "string | number | undefined" depending on what you have passed to `q-header` as `height-hint` prop.
})
</script>
```

You get autocomplete, detailed explanation, and the type information:

![image](https://user-images.githubusercontent.com/6266078/151661363-baef91de-92d7-41ed-959a-5c18afec8bb6.png)
